src/hotspot/share/opto/loopnode.cpp

```diff
@@ -2311,8 +2311,15 @@ const Type* LoopLimitNode::Value(PhaseGVN* phase) const {
     int final_int = (int)final_con;
     // The final value should be in integer range since the loop
     // is counted and the limit was checked for overflow.
-    assert(final_con == (jlong)final_int, "final value should be integer");
-    return TypeInt::make(final_int);
+    // Assert checks for overflow only if all input nodes are ConINodes, as during CCP
+    // there might be a temporary overflow from PhiNodes see JDK-8309266
+    assert(in(Init)->is_ConI() && in(Limit)->is_ConI() && in(Stride)->is_ConI() \
```

Comment:
Do we really need a backslash here?

Reply:
You can wrap lines in asserts without backslashes.

Reply:
Ahh yes. Not sure why, I probably just used the backslash after having read it in some other parts of the code. Sorry for the confusion from my side